File: apps/muly-dotnet-server/src/APIs/Author/Dtos/AuthorUpdateInput.cs
namespace MulyDotnet.APIs.Dtos;

public class AuthorUpdateInput
{
    public string? Id { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public string? FirstName { get; set; }

    public string? LastName { get; set; }

    public DateTime? Dob { get; set; }

    public string? Biography { get; set; }

    public List<BookIdDto>? Books { get; set; }
}

File: apps/muly-node-js-admin/src/api/author/AuthorUpdateInput.ts
import { BookUpdateManyWithoutAuthorsInput } from "./BookUpdateManyWithoutAuthorsInput";

export type AuthorUpdateInput = {
  firstName?: string | null;
  lastName?: string | null;
  dob?: Date | null;
  biography?: string | null;
  books?: BookUpdateManyWithoutAuthorsInput;
};
